app/component/itinerary/BicycleLeg.js

```diff
@@ -52,7 +52,9 @@ export default function BicycleLeg(
   let legDescription = <span>{leg.from ? leg.from.name : ''}</span>;
   const firstLegClassName = index === 0 ? 'start' : '';
   let modeClassName = 'bicycle';
-  const [address, place] = splitStringToAddressAndPlace(leg.from.name);
+  const [name, place] = splitStringToAddressAndPlace(leg.from.name);
+  const address =
+    leg.from.viaLocationType && leg.viaAddress ? leg.viaAddress : name;
```

Comment:
Slightly simpler way (not really important, though):

`address = leg.from.viaLocationType && leg.viaAddress ||  name;`

javascript evaluates && chain to the last value, not to a boolean